torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py

```diff
@@ -0,0 +1,510 @@
+        )
+        output_node.meta["quantization_annotation"] = QuantizationAnnotation(
+            output_qspec=act_qspec,
+            _annotated=True,
+        )
+
+
+def _annotate_add_relu(
+    gm: torch.fx.GraphModule,
+    quantization_config: Optional[QuantizationConfig],
+    filter_fn: Optional[Callable[[Node], bool]] = None,
+) -> None:
+    fused_partitions = find_sequential_partitions(
+        gm, [torch.add, torch.nn.ReLU], filter_fn
+    )
+    for fused_partition in fused_partitions:
+        add_partition, relu_partition = fused_partition
+        if len(relu_partition.output_nodes) > 1:
+            raise ValueError("Relu partition has more than one output node")
+        relu_node = relu_partition.output_nodes[0]
+        if len(add_partition.output_nodes) > 1:
+            raise ValueError("add partition has more than one output node")
+        add_node = add_partition.output_nodes[0]
+
+        if _is_annotated([relu_node, add_node]):
+            continue
+
+        input_act_qspec = get_input_act_qspec(quantization_config)
+        output_act_qspec = get_output_act_qspec(quantization_config)
+
+        input_qspec_map = {}
+        input_act0 = add_node.args[0]
+        if isinstance(input_act0, Node):
+            input_qspec_map[input_act0] = input_act_qspec
+
+        input_act1 = add_node.args[1]
+        if isinstance(input_act1, Node):
+            input_qspec_map[input_act1] = input_act_qspec
+
+        add_node.meta["quantization_annotation"] = QuantizationAnnotation(
+            input_qspec_map=input_qspec_map,
+            _annotated=True,
+        )
+        relu_node.meta["quantization_annotation"] = QuantizationAnnotation(
+            output_qspec=output_act_qspec,
+            _annotated=True,
+        )
+
+
+def _annotate_add(
+    gm: torch.fx.GraphModule,
+    quantization_config: Optional[QuantizationConfig],
+    filter_fn: Optional[Callable[[Node], bool]] = None,
+) -> None:
+    add_partitions = get_source_partitions(
+        gm.graph, [operator.add, torch.add], filter_fn
+    )
+    add_partitions = list(itertools.chain(*add_partitions.values()))
+    for add_partition in add_partitions:
+        add_node = add_partition.output_nodes[0]
+        if _is_annotated([add_node]):
+            continue
+
+        input_act_qspec = get_input_act_qspec(quantization_config)
+        output_act_qspec = get_output_act_qspec(quantization_config)
+
+        input_qspec_map = {}
+        input_act0 = add_node.args[0]
+        if isinstance(input_act0, Node):
+            input_qspec_map[input_act0] = input_act_qspec
+
+        input_act1 = add_node.args[1]
+        if isinstance(input_act1, Node):
+            input_qspec_map[input_act1] = input_act_qspec
+
+        add_node.meta["quantization_annotation"] = QuantizationAnnotation(
+            input_qspec_map=input_qspec_map,
+            output_qspec=output_act_qspec,
+            _annotated=True,
+        )
+
+
+OP_TO_ANNOTATOR = {
```

Comment:
In a follow up diff can you make use of decorator pattern to register annotators?

Another thing is that some of the annotate functions are associated with sequential patterns. e.g. Conv2d_relu. Annotate function for that uses find_sequential_partitions. I am wondering if it makes sense to use that as key of the dict.

e.g. instead of OP_TO_ANNOTATOR["conv2d_relu"] = annotate_fn, it could be
OP_TO_ANNOTATOR[[nn.Conv2d, nn.Relu]] = annotate_conv2d_relu_fn
OP_TO_ANNOTATOR[[nn.Conv2d, functional.Relu]] = annotate_conv2d_relu_fn

and registration decorator can enumerate all possible ways of expressing conv2d + relu pattern (e.g. module x functional)

Reply:
sure I can use decorators.

for the other suggestions:
```
OP_TO_ANNOTATOR[[nn.Conv2d, nn.Relu]] = annotate_conv2d_relu_fn
```
right now when we use it, we'll call OP_TO_ANNOTATOR["conv2d_relu"] and it will annotate all conv2d relu patterns, if we change the key, are you suggesting to change the use to
```
OP_TO_ANNOTATOR[(nn.Conv2d, nn.ReLU)](...)
OP_TO_ANNOTATOR[(nn.Conv2d, functional.relu)](...)
```
while all these calls are doing the same thing?

the implementation of `annotate_conv2d_relu` is already tied to the patterns, so I feel it probably does not make a lot of sense to use this directly, maybe I can add another map to store this information if you want?

like 
```
"conv2d_relu": [[nn.Conv2d, nn.ReLU], [nn.Conv2d, functional.relu]]
```